### PR TITLE
fixed bug where Mutect2 ignored germline variants with alternate representation

### LINKED ISSUE #6972: Mutect2 doesn't filter germline events with alternative representations in gnomad
#6329  Bug Report

### Affected tool(s) or class(es)
Mutect2 / FilterMutectCalls

### Affected version(s)
- [x] Latest public release version [version?]

### Description 

Variants with alternative representations in gnomad are not recognized as being the same as called variants in some cases.  This results in variants that are called and not filtered, but they should be filtered by "germline".

As an example of this, in gnomad the site hg19 16:72991715 is represented as:
16	72991715	.	ACCG	GCCG,*,AGCCGCCG	14986622.13	PASS	AC=33700,10,4;AF=0.83,2.463E-4,9.852E-5

But in M2 it is called as:
A->G

Although ACCG->GCCG and A->G are equivalent, they are not recognized as identical.

#### Expected behavior
Mutect2 should filter these variants as germline.

#### Actual behavior
Mutect2 does not filter these variants.

### PR DESCRIPTION
**Pull request by davidbenjamin**
Closes #6972.

@fleharty here you go.

### CODE REVIEW

**Comment by fleharty** [comment]
@davidbenjamin Just two quick things to look at, then we can merge.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2Engine.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import htsjdk.samtools.SAMFileHeader;
 import htsjdk.samtools.util.Locatable;
+import htsjdk.variant.variantcontext.Allele;
```
[the remark this answers is not in the document]
Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2Engine.java:5]: done

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/filtering/ContaminationFilter.java [hunk 2]
```diff
@@ -60,6 +60,11 @@ public ContaminationFilter(final List<File> contaminationTables, final double co
             final double[] logContaminantLikelihoodPerAllele = new double[alleleFrequencies.length];
             final double[] logOddsOfRealVsContaminationPerAllele = new double[alleleFrequencies.length];
             final double[] posteriorProbOfContaminationPerAllele = new double[alleleFrequencies.length];
+
+            if (altADs.length != alleleFrequencies.length || singleContaminantLikelihoodPerAllele.length != alleleFrequencies.length) {
```
[the remark this answers is not in the document]
Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/filtering/ContaminationFilter.java:64]: whoops, that was for putting in a breakpoint.  Fixed.

**Comment by davidbenjamin** [comment]
@fleharty Back to you.

**Comment by droazen** [comment]
@davidbenjamin Do you still want to get this M2 bug fix merged?

**Comment by fleharty** [comment]
@davidbenjamin Sorry I thought this had been merged.  We should merge this!

**Review by fleharty — APPROVED** [review]

**Pull request merged**